CorsixTH/Lua/dialogs/confirm_dialog.lua

```diff
@@ -37,26 +41,37 @@ function UIConfirmDialog:UIConfirmDialog(ui, text, callback_ok, callback_cancel)
   self:setDefaultPosition(0.5, 0.5)
   self.panel_sprites = app.gfx:loadSpriteTable("QData", "Req04V", true)
   self.white_font = app.gfx:loadFont("QData", "Font01V")
-  self.text = text
   self.callback_ok = callback_ok  -- Callback function to launch if user chooses ok
   self.callback_cancel = callback_cancel -- Callback function to launch if user chooses cancel
+  self.text = text
+
+  local maxLineLengthPixels = 152 -- this should come from somewhere else
+
+  -- Text is varable width, calculate it
+  local lineWidthPixels, textHeight = self.white_font:sizeOf(self.text)
+
+  -- This was previously hard coded to 11, but textHeight is actually 12, one less
+  -- But the graphics appear to be drawn to 13 - one more
+  local textRowHeight = textHeight + 1
 
-  -- Check how "high" the dialog must be
-  local w, h = self.white_font:sizeOf(text)
+  -- Calculate how many text rows we need
+  -- Retail has a fixed size box - but existing behaviour is to calculate size of box
+  -- Leave this as it is
+  local totalLines = math.ceil(lineWidthPixels / maxLineLengthPixels)
+  --local totalLines = 9
 
-  self:addPanel(357, 0, 0)  -- Dialog header
-  local last_y = 22
-  -- Rough estimate of how many rows it will be when drawn.
-  for y = 22, h * (w / 160) * 1.4, 11 do -- Previous value: 136
+  local y = topFrameHeight;
+  local i=0
+  while (i < totalLines) do
```

Comment:
Lua tends to 1 based indexing, and doesn't use brackets around conditions. `for i = 1, totalLines do` would be conventional.

Reply:
Thanks for the advice. This was actually akin to my first version, but it does not pass luacheck
``
  local y = topFrameHeight;                                                      
  for i=1, totalLines do                                                         
      self:addPanel(358, 0, y)  -- Dialog background                             
      y = y + textRowHeight                                                      
  end   
``
Checking confirm_dialog.lua                       1 warning
    confirm_dialog.lua:61:7: (W213) unused loop variable i
Total: 1 warning / 0 errors in 1 file

Any ideas - or can this be overridden?

Reply:
You can override the warning in .luacheck.rc or just calculate 'y' from i within the loop.

Reply:
Thanks for the advice. I'll calculate y from i (although it looks a little strange).

Hope this is OK, let me know if not

Reply:
Oh, sorry: the fix for the luacheck warning is simply to use `_` instead of `i`.

Reply:
Brilliant - that looks a lot nicer. I have checked that in. Thanks